### ISSUE TITLE
cannot connect to infura using lite node

### ISSUE THREAD

**Issue by flyworker** [filecoin-project/lotus#8385]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [X] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
1.4.3
```

### Describe the Bug

cannot connect to infura using lite node
when lite node connect with infura filecoin node with project ID and secret,it shows error listen head changes errored: listenHeadChanges ChainNotify call failed: RPC error (-32004): method 'Filecoin.ChainNotify' not supported in this mode (no out channel support)

### Logging Information

```text
2022-03-28T05:26:37.357-0400    ERROR   events  events/observer.go:58   listen head changes errored: listenHeadChanges ChainNotify call failed: RPC error (-32004): method 'Filecoin.ChainNotify' not supported in this mode (no out channel support)
2022-03-28T05:26:38.358-0400    INFO    events  events/observer.go:69   restarting listenHeadChanges
2022-03-28T05:26:38.418-0400    ERROR   events  events/observer.go:58   listen head changes errored: listenHeadChanges ChainNotify call failed: RPC error (-32004): method 'Filecoin.ChainNotify' not supported in this mode (no out channel support)
2022-03-28T05:26:39.419-0400    INFO    events  events/observer.go:69   restarting listenHeadChanges
2022-03-28T05:26:39.486-0400    ERROR   events  events/observer.go:58   listen head changes errored: listenHeadChanges ChainNotify call failed: RPC error (-32004): method 'Filecoin.ChainNotify' not supported in this mode (no out channel support)
2022-03-28T05:26:40.487-0400    INFO    events  events/observer.go:69   restarting listenHeadChanges
2022-03-28T05:26:40.548-0400    ERROR   events  events/observer.go:58   listen head changes errored: listenHeadChanges ChainNotify call failed: RPC error (-32004): method 'Filecoin.ChainNotify' not supported in this mode (no out channel support)
2022-03-28T05:26:41.548-0400    INFO    events  events/observer.go:69   restarting listenHeadChanges
2022-03-28T05:26:41.602-0400    ERROR   events  events/observer.go:58   listen head changes errored: listenHeadChanges ChainNotify call failed: RPC error (-32004): method 'Filecoin.ChainNotify' not supported in this mode (no out channel support)
2022-03-28T05:26:42.603-0400    INFO    events  events/observer.go:69   restarting listenHeadChanges
2022-03-28T05:26:42.661-0400    ERROR   events  events/observer.go:58   listen head changes errored: listenHeadChanges ChainNotify call failed: RPC error (-32004): method 'Filecoin.ChainNotify' not supported in this mode (no out channel support)
```

**Comment by TippyFlitsUK** [comment]
Hi @flyworker 

Can you please confirm the format of the  `FULLNODE_API_INFO` env var that you are using?

Many thanks!

**Comment by flyworker** [comment]
FULLNODE_API_INFO="https://<project_id>:<project_secret>@filecoin.infura.io" lotus daemon --lite

**Comment by ribasushi** [comment]
Duplicate of https://github.com/filecoin-project/lotus/issues/6111#issuecomment-846114377
Stalled as it needs some `multiaddr` design work to correctly express a basic-auth tuple within an address

If I remember right @frrist was the last one to look into this. Also perhaps @lidel has ideas?

**Comment by flyworker** [comment]
@ribasushi  yes, I think this issue force people to run their own node since no other node provider at the moment?

**Comment by ribasushi** [comment]
@flyworker `api.chain.love` should work:

Here is a howto I wrote somewhere, pasting it here before it makes it properly into some doc ( cc @johnnymatthews )

> to interact with a filecoin hardware wallet securely from the comfort of your own (mac/linux/potentially WSL2) laptop, in about ~5 minutes start-to-finish you need to:
> 
> - `git clone https://github.com/filecoin-project/lotus && cd lotus`
> - `git checkout v1.15.0` ( current stable, number will change as time goes on )
> - `make lotus lotus-shed` ( this might need you installing golang, see https://go.dev/doc/install )
> - `LOTUS_WALLET_ENABLELEDGER=1 FULLNODE_API_INFO=wss://api.chain.love/ ./lotus daemon --lite`
> ( above will keep running on the terminal, open a new one for the commands below )
> - physically connect and unlock your ledger, make sure Ledger Live (the software) is closed ( if you are on linux you might need to do udev shenanigans as per https://support.ledger.com/hc/en-us/articles/115005165269?support=true )
> - `./lotus-shed ledger list`
> - `./lotus-shed ledger key-info  "m/44'/461'/0'/0/{{the-numeric-id-of-the-one-you-want}}"`
> - `./lotus wallet import --format json-lotus` => paste the thing from above on the prompt
> - `./lotus wallet balance`

**Comment by ribasushi** [comment]
@TippyFlitsUK note - while what I gave @flyworker above is a decent workaround, we still need to get lotus to support basic-auth, since this is all Infura offers.

"You can't use lotus-lite with infura" is a bad look.

**Comment by TippyFlitsUK** [comment]
Many thanks @ribasushi !

I have closed the older duplicate and linked to this one as it contains updated workaround info. 

I've bumped this new ticket up to perform triage again on the [ Lotus+Actors Board](https://github.com/orgs/filecoin-project/projects/11#card-80066764) as the original one had been marked as closed in error.

**Comment by lidel** [comment]
> [](https://github.com/filecoin-project/lotus/issues/6111#issuecomment-846114377) Stalled as it needs some `multiaddr` design work to correctly express a basic-auth tuple within an address
> 
> If I remember right @frrist was the last one to look into this. Also perhaps @lidel has ideas?


Protocol attributes (such as user/pass for HTTP basic auth, or SNI for HTTPS)  are not supported by multiaddr.

People asked for this, but nobody had bandwidth or cared enough to spec and implement it.
For *tl;dr* see my 2019 :skull:  summary in https://github.com/multiformats/multiaddr/issues/63#issuecomment-521694427 and proposal in https://github.com/multiformats/multiaddr/issues/87
